File: cpp/secretsanta.cpp
#include <iostream>
#include <cmath>

using namespace std;


int main() {
    int n;
    cin >> n;

    n = n > 1000 ? 1000 : n;

    
    double prob = 0.0;
    double fact = 1.0;
    for(int i = 0; i < n + 1; i++)
    {
        if (i > 0)
        {
            fact *= i;
        }

        prob += pow(-1, i) / fact;
    }
    cout << 1 - prob << endl;
    return 0;
}

File: java/SecretSanta.java
package icpc;
import java.io.*;
import java.util.*;
public class SecretSanta{
	public static void main(String[]args) throws Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));        
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());

		
        double prob = 0;
        double fact = 1.0;
		double val = 1.0;

        for (double i = 0; i <= n; i++) {
            if (i > 0) {
                fact *= i;
            }

			val = Math.pow(1, i) / fact;
			if(Math.abs(val) < 1e-15)
			{
				break;
			}
            prob += val;
        }

        System.out.println(1.0 - prob);
	}
}